design/resource-filter-field-selectors-design.md

```diff
@@ -0,0 +1,158 @@
+  Version        string                          `yaml:"version"`
+  VolumePolicies []VolumePolicy                  `yaml:"volumePolicies"`
+
+  // FieldSelectorPolicies defines a list of policies to enable Kubernetes
+  // resource field selection to support filtering by resource fields.
+  FieldSelectorPolicies []FieldSelectorPolicy    `yaml:"fieldSelectorPolicies"`
+
+  // we may support other resource policies in the future, and they could be added separately
+  // OtherResourcePolicies: []OtherResourcePolicy
+}
+
+import (
+  k8s.io/apimachinery/pkg/fields
+  metav1 k8s.io/apimachinery/pkg/apis/meta/v1
+)
+
+// FieldSelectorPolicy describes the field selection properties used to identify a
+// Kubernetes resource.
+type FieldSelectorPolicy struct {
+  // GroupResource specifies a group and a resource.
+  GroupResource metav1.GroupResource `json:"groupResource,omitempty"`
+
+  // FieldSelectors specifies a field name and its value, typically, implemented
+  // as a map of string-to-string.
+  // If omitted, then all resources of the group will be selected.
+  // +optional
+  FieldSelectors fields.Set `json:"fieldSelectors,omitempty"`
+}
+```
+
+### Code Design
+
+The field selector mechanism eliminates the needs to perform client-side filtering which requires all resources of a particular type to be fetched beforehand.
+The selection criteria can be added as a filter option to the `LIST` requests using the [`metav1.ListOptions`][6] option.
+The `k8s.io/cli-runtime` also offers a [`resource.Builder`][7] type that can be used to convert client-side field selectors into REST requests, to selectively retrieve Kubernetes resources.
+For example, `kubectl get` uses the [following code][8] to instantiate a `resource.Builder` to retrieve pods that satisfy the given fields selection criteria:
+
+```go
+	r := f.NewBuilder().
+		Unstructured().
+		NamespaceParam(o.Namespace).DefaultNamespace().AllNamespaces(o.AllNamespaces).
+		FilenameParam(o.ExplicitNamespace, &o.FilenameOptions).
+		LabelSelectorParam(o.LabelSelector).
+		FieldSelectorParam(o.FieldSelector).
+		Subresource(o.Subresource).
+		RequestChunksOf(chunkSize).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		TransformRequests(o.transformRequests).
+		Do()
+```
+
+## Alternatives Considered
+
+Velero can issue requests to get all resources of a particular type, and then perform client-side field comparisons to find the resource that satisfies a specific field such as resource name. 
+Such retrieval operations on large clusters can be very inefficient.
+The existing label selection filtering mechanism has also been considered.
+But on clusters with many operators, labeling of CRDs requires user to devise a precise labeling scheme to properly group the CRDs, according to operators, which can be error prone.
+
+## Security Considerations
+
+N/A.
+
+## Compatibility
+
+The field selector mechanism only works with the new `ResourcePolicies` API.
```

Comment:
Please add the conflict resolution about the  FieldSelectorPolicy with current existing rules.

such as if the FieldSelectorPolicy selected one resource and Action is `Skip` but the user used --include-resources when backup

Reply:
Is it necessary for field selector policy to support `action`? At the moment, `action` seems to be specific to volume policy. Also, like labels, the K8s field selector API only works with `=` and `!=`. It doesn't support the equivalence of `Skip`.